code/agents.py

```diff
@@ -67,6 +67,8 @@ def calculate_fish_forces(self):
         for (fish, pos) in neighbours:
 
             magnitude = k / np.linalg.norm(pos - self.position)**power
+            if magnitude > 2:
```

Comment:
The way I see it - magnitude is controlled by k and power? I'm guessing a cap on force magnitude could serve some purpose.